### ISSUE TITLE
ERA7B1 vs ERA201B1 Schematic

### ISSUE THREAD

**Issue by pacmac** [lpollier/battmon#3]
Hi;

Firstly, thanks so much for taking the time to reverse engineer the ERA201B1: https://github.com/lpollier/battmon/blob/master/example/ERA-201B1_reverse_engineering_schematic.pdf

I have several ERS7's which use the  ERA7B1 battery pack and have re-celled many of them using my own python script, but I do have a few bad boards. I managed to find a supplier of the fuses and have a few of those, the eeproms, fets and bq2060 are still available but not plentiful.

Looking at your schematic, it does look very similar in design, do you happen to know whether the electronic design of the boards on these 2 packs would be the same ?

**Comment by pacmac** [comment]
OK, silly me, just realised that the ERA7B1 uses a BQ2060 and not a BQ2040, so the answer would be definetely no.

Dont suppose you are aware of any reverse engineering on the ERA7B1 do you ?

**Comment by lpollier** [comment]
Hi Pacmac,

I don't have an ERS-7 Aibo robot in my possession, so yes I can't see how the ERA-7B1 / ERA-7B2 battery PCB is designed.

However, based on the pictures of the PCB found on the internet as well as by noting on some sites that it is possible to read the same characteristics by communicating with the bq2060 as with the bq2040 in the same way, I can say that the Battmon Program 2.0 is fully compatible with ERA-7B1 / ERA-7B2 batteries. But the pinout of the connector may be different betweem them.

It also indicates that the differences with the ERA-7B1 / ERA-7B2 and ERA-201B1 PCBs are not really fundamental to repairing your batteries. My guess would be that you need to set some of the EEPROM memory block values, such as "Cycle Count" to 0 and "Pack Capacity" to the capacity of your new cell battery.
More information about that on the bq2060 datasheet in the "Table 10. EEPROM memory card" section:
https://www.ti.com/lit/ds/symlink/bq2060.pdf

If you have the PCB, you can send me high resolution photos of both sides to this address: loic.pollier at protonmail.com.
So I can verify that there is a battery protection IC like S-8232AAFT on the ERA-201B1 PCB.
Loïc

**Comment by pacmac** [comment]
Thanks very much.

Yes I wrote my own python library on micropython a few years ago, an can reprogram the eeprom and perform resets, cycle count, manufacture date, design capacity etc etc. I have re-celled probably 20 batteries. 

I began to reverse engineer the circuit board a few times, so that I could understand the circuit, but never finished it, maybe I will one day.

Thanks for your reply and excellent work.
